### PR TITLE
Can now select entire flag and correct

### PR DESCRIPTION
**Pull request by robinedman**
Fixes https://github.com/guardian/scribe-plugin-noting/issues/120

We were...
- ... using `isWithinNote` when we should have used `isEntirelyWithinNote`.
- ... not passing `tagName` in a couple of places. (Maybe better to make these mandatory instead of optional?)

### CODE REVIEW

File: src/generate-note-controller.js [hunk 1]
```diff
@@ -223,7 +223,7 @@ module.exports = function(scribe){
         if(markers.length >= 0){
           //check that the selection is within a note
           config.get('selectors').forEach((selector)=>{
-            if(isSelectionEntirelyWithinNote(markers, selector.tagName)){
+            if(isSelectionWithinNote(markers, selector.tagName)){
```
Comment (cutandpastey) [inline src/generate-note-controller.js]: Are you 100% sure this is correct?

**Comment by cutandpastey** [comment]
I think there is more work to be done see: https://github.com/guardian/scribe-plugin-noting/blob/re-select-whole-flag-correct/src/generate-note-controller.js#L130 which does not pass tagName.

File: src/generate-note-controller.js (hunk 1 above)
Comment (robinedman) [inline src/generate-note-controller.js]: It does solve this issue, and all tests pass. But thinking about it maybe you're right. To check that it's entirely within might make more sense.

**Comment by robinedman** [comment]
Ok. What do you think about removing the defaults and making `tagName` mandatory everywhere? It's easy to get this wrong.

**Comment by cutandpastey** [comment]
Dunno, the intention was to reduce errors by falling back to `gu-note` but in practice this ends up causing more problems than it solves.

**Comment by robinedman** [comment]
Let's not do it as part of this PR. I created an issue: #123

**Comment by robinedman** [comment]
That we don't pass a `tagName` to `this.selectNote` is correct though. That function will iterate through the all the tag names.

**Comment by cutandpastey** [comment]
:+1:

File: src/actions/noting/select-note.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+module.exports = function selectNote(noteSegment, tagName = config.get('defaultTagName')){
+  if (!isVFocus(noteSegment)) {
+    errorHandle('Only a valid VFocus element can be passed to selectNote, you passed: %s', noteSegment);
+  }
+
+  var noteSegments = findEntireNote(noteSegment, tagName);
+  removeScribeMarkers(noteSegment);
+
+  noteSegments[0].prependChildren(createVirtualScribeMarker());
```
Comment (cutandpastey) [inline src/actions/noting/select-note.js]: <3

Comment (cutandpastey) [inline src/actions/noting/select-note.js]: Nice!

**Comment by cutandpastey** [comment]
:+1:  I like that you have taken to using the `VFocus` utils!

**Pull request merged**